src/anyio/_core/_fileio.py

```diff
@@ -114,3 +125,346 @@ async def open_file(file: Union[str, PathLike, int], mode: str = 'r', buffering:
+        return await to_thread.run_sync(self._path.read_text, encoding, errors)
+
+    def relative_to(self, *other: Union[str, PathLike]) -> 'Path':
+        return Path(self._path.relative_to(*other))
+
+    async def readlink(self) -> 'Path':
+        target = await to_thread.run_sync(os.readlink, self._path)
+        return Path(cast(str, target))
+
+    async def rename(self, target: Union[str, pathlib.PurePath, 'Path']) -> 'Path':
+        if isinstance(target, Path):
+            target = target._path
+
+        await to_thread.run_sync(self._path.rename, target)
+        return Path(target)
+
+    async def replace(self, target: Union[str, pathlib.PurePath, 'Path']) -> 'Path':
+        if isinstance(target, Path):
+            target = target._path
+
+        await to_thread.run_sync(self._path.replace, target)
+        return Path(target)
+
+    async def resolve(self, strict: bool = False) -> 'Path':
+        func = partial(self._path.resolve, strict=strict)
+        return Path(await to_thread.run_sync(func, cancellable=True))
+
+    def rglob(self, pattern: str) -> AsyncIterator['Path']:
+        gen = self._path.rglob(pattern)
+        return _PathIterator(gen)
+
+    async def rmdir(self) -> None:
+        await to_thread.run_sync(self._path.rmdir)
+
+    async def samefile(self, other_path: Union[str, bytes, int, pathlib.Path, 'Path']) -> bool:
+        if isinstance(other_path, Path):
+            other_path = other_path._path
+
+        return await to_thread.run_sync(self._path.samefile, other_path, cancellable=True)
+
+    async def stat(self, *, follow_symlinks: bool = True) -> os.stat_result:
+        func = partial(os.stat, follow_symlinks=follow_symlinks)
+        return await to_thread.run_sync(func, self._path, cancellable=True)
+
+    async def symlink_to(self, target: Union[str, pathlib.Path, 'Path'],
+                         target_is_directory: bool = False) -> None:
+        if isinstance(target, Path):
+            target = target._path
+
+        await to_thread.run_sync(self._path.symlink_to, target, target_is_directory)
+
+    async def touch(self, mode: int = 0o666, exist_ok: bool = True) -> None:
+        await to_thread.run_sync(self._path.touch, mode, exist_ok)
+
+    async def unlink(self, missing_ok: bool = False) -> None:
+        try:
+            await to_thread.run_sync(self._path.unlink)
+        except FileNotFoundError:
+            if not missing_ok:
+                raise
+
+    def with_name(self, name: str) -> 'Path':
+        return Path(self._path.with_name(name))
+
+    def with_stem(self, stem: str) -> 'Path':
+        return Path(self._path.with_name(stem + self._path.suffix))
+
+    def with_suffix(self, suffix: str) -> 'Path':
+        return Path(self._path.with_suffix(suffix))
+
+    async def write_bytes(self, data: bytes) -> int:
+        return await to_thread.run_sync(self._path.write_bytes, data)
+
+    async def write_text(self, data: str, encoding: Optional[str] = None,
```

Comment:
Why not to_thread the write_text directly?

Reply:
Because on py < 3.10, the `newline` parameter doesn't exist.

Reply:
Would it be better to do this in one call to to_thread?

Reply:
Done